test/src/test/java/hudson/cli/DeleteBuildsCommandTest.java

```diff
@@ -139,24 +136,18 @@ public class DeleteBuildsCommandTest {
         assertThat(result.stdout(), containsString("Deleted 0 builds"));
     }
 
-    @Test public void deleteBuildsShouldSuccessEvenTheBuildIsRunning() throws Exception {
```

Comment:
I did not notice this test before. It was added in https://github.com/jenkinsci/jenkins/pull/2310 along with all of the other tests for `delete-builds`.

I still think the previous behavior was undesirable - deleting builds that are still running and reachable in memory does not make sense, can lead to tons of errors in the Jenkins logs, and any builds deleted while running are very likely to come back after a Jenkins restart given that build completion will cause `build.xml` to be rewritten.

I can see the argument though that `Run.delete` should work like `Job.delete` and cancel the build if it is running, waiting up to 15 seconds and only throwing an exception if the build still hasn't completed at that point. @jglick suggested this approach as well, but I was hoping to avoid adding complexity here unless we really think that behavior is necessary. Any opinions on this?

Reply:
No strong opinion. I think the proposed error behavior is intuitive enough. You can explicitly abort the build first if that is what you intended.